.github/workflows/flutter_vm.yaml

```
@@ -82,6 +86,14 @@ jobs:
         working-directory: ${{ inputs.working-directory }}
 
       - name: Run Tests
+        id: testJob
         if: "always() && steps.bootstrap.conclusion == 'success' && steps.testCheck.outputs.hasTests == 'true'"
         run: flutter test
         working-directory: ${{ inputs.working-directory }}
+
+      - name: Archive golden failures
+        if: "always() && inputs.has-goldens && steps.testJob.conclusion == 'failure'"
+        uses: actions/upload-artifact@65d862660abb392b8c4a3d1195a2108db131dd05
```

Add the version or branch in a comment at the end
```suggestion
        uses: actions/upload-artifact@a8a3f3ad30e3422c9c7b888a15615d19a852ae32 # 3.1.3
```